arrows/darknet/darknet_trainer.cxx

```diff
@@ -290,7 +284,7 @@ train_from_disk(
 #else
   std::string darknet_cmd = "darknet";
 #endif
-  std::string darknet_args = "-i " + boost::lexical_cast< std::string >( d->m_gpu_index ) +
+  std::string darknet_args = "-i " + std::to_string( d->m_gpu_index ) +
```

Comment:
Rather OT, but I am somewhat alarmed that we are invoking this in a way that doesn't take a vector of arguments...

Reply:
I'm not sure what you mean by this comment. Can you explain with more detail? I would be interested in correcting most anything that causes alarm.

Reply:
Argument splitting is typically a shell function. At the OS level, you usually want to start processes with the arguments already split. For example, in Python, this might look like:

```python
  cmd = 'darknet'
  args = ['-i', str(gpu_index), 'detector', 'train',
          train_directory + '/yolo_v2.data,
          train_directory + '/yolo_v2.cfg,]
  # ...
  popen(cmd, args, ...)
```

Note that `args` is *an array* of strings, not a single string.

By using `system` instead of e.g. `execvp`, you are opening yourself up to the command string being subject to additional parsing. For example, if `d->m_train_directory` contains spaces, quotes or other special characters, something unintended may happen.

Alas, Windows is a PITA here, with only CreateProcess (which accepts only a string) "fully" supported. (OTOH, I don't think we care about UWP.)

Reply:
Understood now, thanks! I definitely agree with you on that. I might take a look into that issue at some point. I'm on a bit of a mission to go through kwiver and find/fix issues like this and clean it up where it needs. Mostly I just wanted to make sure you weren't speaking about something directly related to my changes.

Reply:
> Mostly I just wanted to make sure you weren't speaking about something directly related to my changes.

Nope, that's literally what the "OT" (ahem: "Off Topic" if you didn't know) means in this context.